staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go

```diff
@@ -131,6 +133,24 @@ func (IntOrString) OpenAPISchemaType() []string { return []string{"string"} }
 // the OpenAPI spec of this type.
 func (IntOrString) OpenAPISchemaFormat() string { return "int-or-string" }
 
+// OpenAPIV3Definition is the customized definition only for OpenAPI v3.
+func (IntOrString) OpenAPIV3Definition() openapi.OpenAPIDefinition {
+	return openapi.OpenAPIDefinition{
+		Schema: spec.Schema{
+			SchemaProps: spec.SchemaProps{
+				Description: "IntOrString is a type that can hold an int32 or a string." +
```

Comment:
The full over-write makes me sad, can we avoid having to overwrite the description too, since that makes one more place to update it. Also we have, at least for now, only very few types where we need to do that so maybe it's not the most important.

Reply:
Maybe instead of completely overriding the full definition, we could have a method that only copies non-nil values?

Reply:
That's one way of doing it. However, in our use case, we are defining a type. There is hardly any other use cases of just setting its OpenAPI type(s). Allowing overwriting anything will be a major overkill.